src/Digdir.Domain.Dialogporten.Infrastructure/Altinn/ResourceRegistry/ResourceRegistryClient.cs

```diff
@@ -99,21 +100,27 @@ private async Task<ServiceResourceInformation[]> FetchServiceResourceInformation
     {
         const string searchEndpoint = "resourceregistry/api/v1/resource/resourcelist";
 
-        var response = await _client
-            .GetFromJsonEnsuredAsync<List<ResourceListResponse>>(searchEndpoint,
-                cancellationToken: cancellationToken);
-
-        return response
-            .Where(x => !string.IsNullOrWhiteSpace(x.HasCompetentAuthority.Organization))
-            .Where(x => x.ResourceType is
-                ResourceTypeGenericAccess or
-                ResourceTypeAltinnApp or
-                ResourceTypeCorrespondence)
-            .Select(x => new ServiceResourceInformation(
-                $"{Constants.ServiceResourcePrefix}{x.Identifier}",
-                x.ResourceType,
-                x.HasCompetentAuthority.Organization!))
-            .ToArray();
+        return await _cache.GetOrSetAsync(
+            ServiceResourceInformationCacheKey,
+            async cToken =>
+            {
+                var response = await _client
+                    .GetFromJsonEnsuredAsync<List<ResourceListResponse>>(searchEndpoint,
+                        cancellationToken: cToken);
```

Comment:
I think this can help with keeping these two in sync.

 * ServiceResourceInformationByOrgCacheKey expires, fetches the cached raw response
 * This raw response then expires
 * ServiceResourceInformationByResourceIdCacheKey expires, fetches the raw response from the remote (which could have changed in the meanwhile)
 * ServiceResourceInformationByOrgCacheKey and ServiceResourceInformationByResourceIdCacheKey are now out of sync for the expiry duration

So whenever the raw expires, invalidate the two that depend on it?  
 
```suggestion
                        cancellationToken: cToken);

                await _cache.ExpireAsync(ServiceResourceInformationByOrgCacheKey, token: cToken);
                await _cache.ExpireAsync(ServiceResourceInformationByResourceIdCacheKey, token: cToken);
```